### PR TITLE
fix freetype apothecary script, and therefore cairo

### PR DESCRIPTION
**Pull request by tgfrerer**
In the OS X recipe, freetype's pkgconfig entry did not point to the correct fat static library, and would therefore scuttle the cairo build.

This pr fixes this by copying the fat static freeimage lib to the point indicated by freeimage's pkgconfig .pc file, so that the build can proceed properly. Also enables multi-processor builds to save time drinking tea.

## builds FAT libs for cairo, freetype:
* 32bit libstdc++
* 64bit libc++

This should get us one step closer to tackling #2016

### CODE REVIEW

**Comment by kylemcdonald** [comment]
woohoo! cc @openframeworks/apothecary

**Comment by danoli3** [comment]
Just saw you mention FreeImage however it is FreeType script that is
modified, is that correct?

I have accidentally labelled a PR FreeImage for FreeType before so just
aware of this ^.^

**Comment by tgfrerer** [comment]
@danoli3 bang on =) ! thanks for spotting this, t'was indeed freetype, i've updated the title

**Comment by danoli3** [comment]
Cairo: [Build Status](https://travis-ci.org/danoli3/apothecary-den) :+1: 
FreeType: [Build Failing](https://travis-ci.org/danoli3/apothecary-den/builds/49261696)

Problem:
```cp: directory build/_buildroot/lib/pkgconfig does not exist```

File: scripts/apothecary/formulas/freetype.sh [hunk 1]
```diff
@@ -73,14 +73,24 @@ function build() {
...
 
+
+
 		cd lib/$TYPE/
 		lipo -create libfreetype-i386.a \
 					libfreetype-x86_64.a \
 					-output libfreetype.a
+
+		# copy pkgconfig file to _buildroot/lib/pkgconfig
+		cp -v $BUILD_TO_DIR/lib/pkgconfig/freetype2.pc $BUILD_ROOT_DIR/lib/pkgconfig/
```
Comment (danoli3) [inline scripts/apothecary/formulas/freetype.sh]: You're going to have to add a ```mkdir -p "$BUILD_ROOT_DIR/lib/pkgconfig/" ``` before the ```cp```

**Comment by danoli3** [comment]
Copy the files changed, to outside the repo: Then:
```git reset --hard SHA``` where SHA is the HEAD of the openframeworks master (will reset commit)
```git push -f``` (force push to your origin/branch).
Then re-open this pull request on here (github keep closing them automatically now at this point, this will prevent "infinity" commits).
Then copy your changes back from your local files, fix the mkdir thingo and recommit to your branch/origin. XD

**Comment by danoli3** [comment]
Sent @tgfrerer a pull request to fix this then all should be good to merge.

**Comment by tgfrerer** [comment]
Thanks, Daniel, i've merged your PR into this PR, I think we're ready to merge this PR =)

**Comment by danoli3** [comment]
[Build Status](https://travis-ci.org/danoli3/apothecary-den)

:+1:

**Comment by tgfrerer** [comment]
\o/

**Comment by arturoc** [comment]
hey, sorry to be so picky :) but perhaps we should have the -j as a global option to apothecary instead of the global formulas. we could have a $(MAKE) instead of make and overwrite that from the main script to be -j but for now i would remove the -j to be consistent to everything else. also having the option is nice since some times you don't want the compiling to eat up all the resources in the computer

**Comment by danoli3** [comment]
Also the ```-j``` has issues compiling some libraries such as OpenSSL which relies on non-concurrent building. :+1: for global on this (however must have a way to override it for a lib that cannot use it).

**Comment by bilderbuchi** [comment]
:+1: on arturo's `-j` suggestion.

**Pull request merged**